src/taipy/core/pipeline/_pipeline_manager.py

```diff
@@ -124,6 +126,8 @@ def _hard_delete(cls, pipeline_id: PipelineId):
     def _get_owned_entity_ids(cls, pipeline: Pipeline) -> _EntityIds:
         entity_ids = _EntityIds()
         for task in pipeline.tasks.values():
+            if not isinstance(task, Task):
+                task = _TaskManagerFactory._build_manager()._get(task)
```

Comment:
How can task variable can be a task ID here?

Reply:
Not sure but I check the type of `pipeline.tasks` and it returns a list of task ids or tasks.

```python
    @tasks.setter  # type: ignore
    @_self_setter(_MANAGER_NAME)
    def tasks(self, tasks: Union[List[TaskId], List[Task]]):
        self._tasks = tasks
```